src/test/kotlin/org/dif/crypto/key/RecipientKeySelectorTest.kt

```diff
@@ -0,0 +1,221 @@
+import org.dif.exceptions.IncompatibleCryptoException
+import org.dif.exceptions.SecretNotFoundException
+import org.dif.fixtures.JWM
+import org.dif.mock.BobSecretResolverMock
+import org.dif.mock.DIDDocResolverMock
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+class RecipientKeySelectorTest {
+    @Test
+    fun `Test find verification key`() {
+        val recipientKeySelector = RecipientKeySelector(DIDDocResolverMock(), BobSecretResolverMock())
+        val expected = "did:example:alice#key-2"
+        val key = recipientKeySelector.findVerificationKey(expected)
+        assertEquals(expected, key.id)
+    }
+
+    @Test
+    fun `Test find anon crypto keys`() {
+        val recipientKeySelector = RecipientKeySelector(DIDDocResolverMock(), BobSecretResolverMock())
+
+        val expected = listOf(
+            "did:example:bob#key-x25519-1",
+            "did:example:bob#key-x25519-2",
+            "did:example:bob#key-x25519-3"
+        )
+
+        val keys = recipientKeySelector.findAnonCryptKeys(expected)
+        assertContentEquals(expected, keys.toList().map { it.id })
+    }
+
+    @Test
+    fun `Test find second anon crypto key`() {
+        val recipientKeySelector = RecipientKeySelector(DIDDocResolverMock(), BobSecretResolverMock())
+
+        val recipient = listOf(
+            "did:example:bob#key-x25519-4",
+            "did:example:bob#key-x25519-2",
+            "did:example:bob#key-x25519-5"
+        )
+
+        val expected = listOf(
+            "did:example:bob#key-x25519-2"
+        )
+
+        val keys = recipientKeySelector.findAnonCryptKeys(recipient)
+        assertContentEquals(expected, keys.map { it.id }.toList())
+    }
+
+    @Test
+    fun `Test find auth crypto keys`() {
+        val recipientKeySelector = RecipientKeySelector(DIDDocResolverMock(), BobSecretResolverMock())
+
+        val sender = "did:example:alice#key-x25519-1"
+        val recipient = listOf(
+            "did:example:bob#key-x25519-1",
+            "did:example:bob#key-x25519-2",
+            "did:example:bob#key-x25519-3"
+        )
+
+        val (from, to) = recipientKeySelector.findAuthCryptKeys(sender, recipient)
+
+        val expected = Pair(
+            "did:example:alice#key-x25519-1",
+            listOf(
+                "did:example:bob#key-x25519-1",
+                "did:example:bob#key-x25519-2",
+                "did:example:bob#key-x25519-3"
+            )
+        )
+
+        assertEquals(expected.first, from.id)
+        assertContentEquals(expected.second, to.toList().map { it.id })
+    }
+
+    @Test
+    fun `Test DID is passed to methods`() {
+        val recipientKeySelector = RecipientKeySelector(DIDDocResolverMock(), BobSecretResolverMock())
+        val expected = "'DID URL' was excepted"
+
+        run {
+            val actual = assertFailsWith<IllegalStateException> {
```

Comment:
I believe `IllegalStateException` must be wrapped by a DIDCommException

Reply:
Let's fix it next PR